### PR TITLE
Generic filtering

### PR DESCRIPTION
**Pull request by ryandillinfelton**

### CODE REVIEW

**Comment by ryandillinfelton** [comment]
Converting to draft while I update the readme

**Review by kruai — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -13,11 +13,30 @@ the actual System Profile shape.
 Document version is reflected in the filename (_v1.yaml_) and under _$version_ root key. _$id_ is a file name used in
 the Host Inventory.
 
+## Contributing ##
+
+### System Profile ###
+
+When contributing a new field to the system_profile schema please insure you complete the following steps:
```
Comment (kruai) [inline README.md:20]:
```suggestion
When contributing a new field to the system_profile schema, please ensure you complete the following steps:
```

File: schemas/system_profile/v1.yaml [hunk 2]
```diff
@@ -279,6 +282,8 @@ $defs:
           $ref: '#/$defs/InstalledProduct'
       insights_client_version:
         type: string
+        description: the version number of insights client. supports wildcards
```
Comment (tahmidefaz) [inline schemas/system_profile/v1.yaml:285]: Lint is failing because of this line. You need to change it.

**Review by tahmidefaz — CHANGES_REQUESTED** [review]
Looks good! Just check the lint. I think just capitalizing the first letters of the sentences will fix it.

File: schemas/system_profile/v1.yaml (hunk 2 above)
Comment (ryandillinfelton) [reply schemas/system_profile/v1.yaml]: All checks passing

**Review by kruai — APPROVED** [review]
LGTM!

**Review by tahmidefaz — APPROVED** [review]
Lookin' good! :+1:

**Pull request merged**